packages/ag-charts-community/src/components/toolbar/toolbar.ts

```diff
@@ -0,0 +1,177 @@
+        }
+    }
+
+    public toggleButtonEnabledByIndex(index: number, enabled: boolean) {
+        const buttonWidget = this.buttonWidgets.at(index);
+        if (!buttonWidget) return;
+        buttonWidget.getElement().ariaDisabled = (!enabled).toString();
+    }
+
+    public toggleSwitchCheckedByIndex(index: number, checked: boolean) {
+        const buttonWidget = this.buttonWidgets.at(index);
+        if (!buttonWidget) return;
+        buttonWidget.getElement().ariaChecked = checked.toString();
+    }
+
+    public toggleButtonVisibilities(visibleIndices: Array<number>) {
+        for (const [index, buttonWidget] of this.buttonWidgets.entries()) {
+            buttonWidget
+                .getElement()
+                .classList.toggle('ag-charts-toolbar__button--hidden-toggled', !visibleIndices.includes(index));
+        }
+        this.refreshButtonClasses();
+    }
+
+    public getButtonBounds() {
+        return this.buttonWidgets.map((buttonWidget) => {
+            const element = buttonWidget.getElement();
+            const parent = element.offsetParent as HTMLElement | null;
+            return new BBox(
+                element.offsetLeft + (parent?.offsetLeft ?? 0),
+                element.offsetTop + (parent?.offsetTop ?? 0),
+                element.offsetWidth,
+                element.offsetHeight
+            );
+        });
+    }
+
+    private refreshButtonClasses() {
+        const { buttonWidgets, onDragStart } = this;
+
+        let first: boolean;
+        let last: boolean;
+        let section: string | null | undefined;
+
+        for (const [index, buttonWidget] of buttonWidgets.entries()) {
+            const element = buttonWidget.getElement();
+
+            first = (onDragStart == null && index === 0) || section != element.getAttribute('data-section');
+            last =
+                index === buttonWidgets.length - 1 ||
+                element.getAttribute('data-section') !=
+                    buttonWidgets
+                        .at(index + 1)
+                        ?.getElement()
+                        .getAttribute('data-section');
+
+            element.classList.toggle('ag-charts-toolbar__button--first', first);
+            element.classList.toggle('ag-charts-toolbar__button--last', last);
+            element.classList.toggle('ag-charts-toolbar__button--gap', index > 0 && first);
+
+            section = element.getAttribute('data-section');
+        }
+    }
+
+    private createDragHandle() {
+        const { onDragStart } = this;
+        if (!onDragStart) return;
+
+        const dragHandle = new NativeWidget<HTMLElement>(createElement('div', 'ag-charts-toolbar__drag-handle'));
+        dragHandle.getElement().innerHTML = `<span class="${getIconClassNames('drag-handle')} ag-charts-toolbar__icon"></span>`;
+        dragHandle.getElement().addEventListener('mousedown', (event) => {
+            onDragStart(event, dragHandle.getElement());
+        });
```

Comment:
I may add an `IconWidget` at some point, and then it can just be `dragHandle.appendChild(new IconWidget('drag-handle'))`.

I also looked at adding `MouseWidgetEvent<'mousedown'>` but that percolated through to the region manager and it needed to be converted to a `'click'` event here - https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-community/src/chart/interaction/regionManager.ts#L179 - which seemed a little risky? So I've left this for now.

Reply:
That's fine. Leave it as is for now.